iidm/iidm-api/src/main/java/com/powsybl/iidm/network/identifiers/IdWithWildcardsNetworkElementIdentifier.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.identifiers;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.Network;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.network.identifiers.NetworkElementIdentifier.IdentifierType.ID_WITH_WILDCARDS;
+
+/**
+ *
+ * Identifier that finds a network element that have some unknown character.
+ * The character '?' is the wildcard that replace the unknown characters
+ * There can be maximum 5 wildcards in the identifier.
+ * For example the identifier "GEN_?" allows to find GEN_1
+ * the only special characters allowed for the identifier are '_', '-', '.'
+ *
+ * @author Etienne Lesot {@literal <etienne.lesot at rte-france.com>}
+ *
+ */
```

Comment:
Here is a suggestion to fix some typos and have a better looking javadoc:
```suggestion
/**
 * <p>Identifier that finds a network element that have some unknown characters.</p>
 * <p>The unknown characters should be replaced in the identifier by the wildcard character '?'.</p>
 * <p>There can be a maximum of 5 wildcards in the identifier and the only special characters allowed for the identifier are '_', '-' and '.'.</p>
 * <p>For example, the identifier "GEN_?" allows to find "GEN_1".</p>
 *
 * @author Etienne Lesot {@literal <etienne.lesot at rte-france.com>}
 */
```